### PR TITLE
OCPBUGS-29208: Fix for ACM Policy Generator Plugin

### PR DESCRIPTION
**Pull request by edcdavid**
For ConfigurationPolicy kind, the name of the object and the inner template are the same when generated with ACM Gen.
In comparison the PGT plugin, adds "-config" to the name of the inner template (see https://github.com/openshift-kni/cnf-features-deploy/blob/dc64c970fa59dacf387f1e432d6a46ea551dbb1c/ztp/policygenerator/policyGen/policyBuilder.go#L422). this leads to overloaded keys in the
clusterGroupUpgrade.Status.SafeResourceNames map.

Changes:
 - namespace+"/"+name key used in the clusterGroupUpgrade.Status.SafeResourceNames map
 - use of namespace instead of namespace length in GetSafeResourceName
 - use of utf8.RuneCountInString() instead of len to support non english characters (see policy web hook at https://github.com/stolostron/governance-policy-propagator/blob/feb9abb241954d7ce64054d7c621b423deece8cf/api/v1/policy_webhook.go#L86-L87 ).
 - adding logs to display the safenames and lenghts

### CODE REVIEW

File: controllers/utils/common.go [hunk 1]
```diff
@@ -32,25 +34,37 @@ func GetMinOf3(number1, number2, number3 int) int {
...
-		safeName = NewSafeResourceName(name, clusterGroupUpgrade.GetAnnotations()[NameSuffixAnnotation], maxLength, spareLength)
-		clusterGroupUpgrade.Status.SafeResourceNames[name] = safeName
+		safeName = NewSafeResourceName(name, namespace, clusterGroupUpgrade.GetAnnotations()[NameSuffixAnnotation], maxLength, log)
+		clusterGroupUpgrade.Status.SafeResourceNames[namespace+"/"+name] = safeName
+	} else {
+		if log != nil {
+			log.Info("safename already in  clusterGroupUpgrade.Status.SafeResourceNames",
+				"safename", safeName,
+				"namespace", namespace,
+				"safeName+namespace length <= 62", utf8.RuneCountInString(safeName+namespace))
```
Comment (jc-rh) [inline controllers/utils/common.go:51]: Nit, if we have to keep this log, drop the "<= 62" part.

File: controllers/utils/common.go [hunk 2]
```diff
@@ -59,12 +73,22 @@ func NewSafeResourceName(name, suffix string, maxLength, spareLength int) string
...
 	}
 
 	// The newSafeResourceName should match regex
 	// `[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*` as per
 	// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
-	return fmt.Sprintf("%s-%s", base, suffix)
+
+	safename = fmt.Sprintf("%s-%s", base, suffix)
+
+	if log != nil {
```
Comment (jc-rh) [inline controllers/utils/common.go:86]: I would drop this log for sure.

File: controllers/utils/constants.go [hunk 3]
```diff
@@ -41,7 +41,9 @@ const (
 
 // CR name length limits and suffix annotation
 const (
-	MaxPolicyNameLength    = 63
+	// Maximum length of policy + namespace (not including extra separator dot so 63 -1 = 62)
+	// this is calculated with utf8.RuneCountInString(policy.Name)+utf8.RuneCountInString+utf8.RuneCountInString(policy.Namespace)
+	MaxPolicyNameLength    = 62
```
Comment (jc-rh) [inline controllers/utils/constants.go:46]: I prefer keep the same value here and count the dot in the length calculation. Or you want to rename it to maxPolicyNameLengthExcludingTheDot?

File: controllers/utils/common.go (hunk 1 above)
Comment (edcdavid) [reply controllers/utils/common.go]: fixed

File: controllers/utils/common.go (hunk 2 above)
Comment (edcdavid) [reply controllers/utils/common.go]: fixed

File: controllers/utils/constants.go (hunk 3 above)
Comment (edcdavid) [reply controllers/utils/constants.go]: fixed

**Pull request merged**